.github/shared-actions/detect-package-manager/action.yaml

```diff
@@ -0,0 +1,68 @@
+name: Detect Package Manager
+author: artur.wolny@frontegg.com
+description: Detect package manager and version using footprints like package-lock.json, yarn.lock, .yarnrc.yml or package.json
+
+outputs:
+  manager:
+    description: Detected package manager
+    value: ${{ steps.pm_detection_results.outputs.manager }}
+  version:
+    description: Detected package manager version
+    value: ${{ steps.pm_detection_results.outputs.version }}
+
+runs:
+  using: composite
+  steps:
+    - name: Check if .yarnrc.yml exists
+      id: yarnrc_yaml_exists
+      uses: andstor/file-existence-action@v2
+      with:
+        files: .yarnrc.yml
+    - name: Check if yarn.lock exists
+      id: yarn_lock_exists
+      uses: andstor/file-existence-action@v2
+      with:
+        files: yarn.lock
+    - name: Check if package-lock.json exists
+      id: package_lock_exists
+      uses: andstor/file-existence-action@v2
+      with:
+        files: package-lock.json
+    - name: Get package-manager from package.json
+      id: package_json_versions
+      shell: bash
+      run: |
+        ACTION_OUTPUT=$(cat package.json | jq '.packageManager')
+        echo $ACTION_OUTPUT | sed -rn 's/^"([^@]*)@(.*)"$/manager=\1/p' >> $GITHUB_OUTPUT
+        echo $ACTION_OUTPUT | sed -rn 's/^"([^@]*)@(.*)"$/version=\2/p' >> $GITHUB_OUTPUT
+    - name: Get version from .yarnrc.yml
+      id: yarnrc_yaml_version
+      if: ${{ steps.yarnrc_yaml_exists.outputs.files_exists == 'true' }}
+      shell: bash
+      run: echo "version=$(cat .yarnrc.yml | yq '.yarnPath' | sed -nr 's/^.*\/yarn-(.+)\.c?js$/\1/p')" >> $GITHUB_OUTPUT
+    - name: Show results
+      shell: bash
+      run: |
+        echo "package_manager(package.json)=${{ steps.package_json_versions.outputs.manager }}"
+        echo "version(package.json)=${{ steps.package_json_versions.outputs.version }}"
+        echo "version(.yarnrc.yml)=${{ steps.yarnrc_yaml_version.outputs.version }}"
+        echo "package_lock_exists=${{ steps.package_lock_exists.outputs.files_exists }}"
+        echo "yarnrc_yaml_exists=${{ steps.yarnrc_yaml_exists.outputs.files_exists }}"
+        echo "yarn_lock_exists=${{ steps.yarn_lock_exists.outputs.files_exists }}"
+    - name: Conclude package manager
+      id: pm_detection_results
+      shell: bash
+      env:
+        YARN_FILES_EXISTS: ${{ steps.yarnrc_yaml_exists.outputs.files_exists == 'true' || steps.yarn_lock_exists.outputs.files_exists == 'true' }}
```

Comment:
In the long run i think we need a more agnostic way of doing this

Reply:
in the long run we should stick to one common package manager, `yarn classic` is not yet supported and we should move away from it

Reply:
> What if you have yarn.lock but not yarn.json because someone forgot ?

There is no `yarn.json` so I suppose you mean `.yamlrc.yml`?

Anyway, if you have `yarn.lock` but don't have `.yamlrc.yml` then the concluded result will be `yarn` with version `1.19.0`